src/meltano/core/plugin/singer/tap.py

```diff
@@ -615,3 +619,58 @@ def catalog_cache_key(self, plugin_invoker):
         key_json = json.dumps(key_dict)
 
         return sha1(key_json.encode()).hexdigest()  # noqa: S303 S324
+
+    def warn_property_not_found(  # noqa: C901
+        self,
+        rules: list[MetadataRule],
+        catalog: CatalogDict,
+    ):
+        """Validate MetadataRules conforms to discovered Catalog.
+
+        Validate MetadataRules against the tap's discovered Catalog & emit
+        warnings for each rule that does not match any property in catalog.
+
+        The method is intentionally written in a defensive manner so as not to
+        raise exceptions. e.g.
+            * use dict.get rather than []-accessor
+            * filter list comprehensions to remove some elements that should
+              not be used for validation
+
+        Args:
+            rules: List of `MetadataRule`
+            catalog: Discovered Source Catalog
+        """
+        property_rules = [
+            r
+            for r in rules
+            if r.tap_stream_id != "*"
+            and len(r.breadcrumb) >= 2
+            and r.breadcrumb != ["properties", "*"]
```

Comment:
I think this is causing select rules like `my_stream.*` to fall through, which should raise a ``Stream `my_stream` was not found`` warning.

Reply:
I think if the condition `r.breadcrumb != ["properties", "*"]` is moved within the for-loop just a few lines below, we can test for non existent streams. changing this.